doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,606 @@
+settings to create the tab we'll hosting the commandline client. If the user has
+`"elevate": true` in their `profiles.defaults`, we'd usually try to
+auto-elevate the profile. In this scenario, however, we can't do that. The
+Terminal is being invoked on behalf of the client app launching, instead of the
+Terminal invoking the client application.
+
+**2021-08-17 edit**: Now that "defterm" has shipped, we're a little more aware
+of some of the limitations with packaged COM and elevation boundaries. Defterm
+cannot be used with elevated processes _at all_ currently (see [#10276]). When
+an elevated commandline application is launched, it will always just appear in
+`conhost.exe`. Furthermore, An unelevated peasant can't communicate with an
+elevated monarch so we can't toss the connection to the elevated monarch and
+have them handle it.
+
+The simplest solution here is to just _always_ ignore the `elevate` property for
+incoming defterm connections. This is not an ideal solution, and one that we're
+willing to revisit if/when [#10276] is ever fixed.
+
+### Elevation on OneCore SKUs
+
+This spec proposes using `ShellExecute` to elevate the Terminal window. However,
+not all Windows SKUs have support for `ShellExecute`. Notably, the non-Desktop
+SKUs, which are often referred to as "OneCore" SKUs. On these platforms, we
+won't be able to use `ShellExecute` to elevate the Terminal. There might not
+even be the concept of multiple elevation levels, or different users, depending
+on the SKU.
+
+Fortunately, this is a mostly hypothetical concern for the moment. Desktop is
+the only publicly supported SKU for the Terminal currently. If the Terminal ever
+does become available on those SKUs, we can use these proposals as mitigations.
+
+* If elevation is supported, there must be some other way of elevating a
+  process. We could always use that mechanism instead.
+* If elevation isn't supported (I'm thinking 10X is one of these), then we could
+  instead display a warning dialog whenever a user tries to open an elevated
+  profile.
+  - We could take the warning a step further. We could add another settings
+    validation step. This would warn the user if they try to mark any profiles
+    or actions as `"elevate":true`
+
+## Future considerations
+
+* If we wanted to go even further down the visual differentiation route, we
+  could consider allowing the user to set an entirely different theme ([#3327])
+  based on the elevation state. Something like `elevatedTheme`, to pick another
+  theme from the set of themes. This would allow them to force elevated windows
+  to have a red titlebar, for example.
+* Over the course of discussion concerning appearance objects ([#8345]), it
+  became clear that having separate "elevated" appearances defined for
+  `profile`s was overly complicated. This is left as a consideration for a
+  possible future extension that could handle this scenario in a cleaner way.

```

Comment:
Can we have an issue to track the discussion on "elevated" vs "unfocused" appearance configs?

I'm wondering if we could just do the following:
|  | Focused | Unfocused |
|--|--|--|
| Regular | regular settings | `unfocusedAppearance` |
| Elevated | `elevatedAppearance` | `elevatedAppearance` |

Reasoning: "elevated" is a long-term state whereas focus constantly shifts

idk, either way, a thread on this might be nice to have.

Reply:
I mean, we might as well just use #1939 for that, yea?